### DISCUSSION TITLE
CollectionConfig removes indexes field

### DISCUSSION THREAD

**Discussion by blueMountain007** [payloadcms/payload#4601] [General]
### Describe the Bug

The indexes field has been removed from CollectionConfig in the latest version, but I can't find any description on how to migrate it in the documentation?

### To Reproduce

When upgrading from v1.11.1 to v2.2.2, I found that the previously supported indexes field is no longer supported.
![image](https://github.com/payloadcms/payload/assets/16400882/3bb3f75c-17bd-4bfc-811b-162f19194fd2)

### Payload Version

2.2.2

**Comment by DanRibbens** [comment]
We removed this collection config when we added database adapters. Since the type was based on MongoDB it didn't make sense to keep it there since it wouldn't work for other adapters.

The recommended way to handle this now is to add indexes inside of migrations.

This is something we should document and maybe we can revisit adding the `indexes` functionality back into the database adapters.

What do you think?

**Comment by blueMountain007** [comment]
@DanRibbens  In order to migrate to the latest version (v2.2.2), how to adjust the code of 'indexes' for the previous v1.11.0 version? Thanks!

**Comment by DanRibbens** [comment]
Hey @liulian-13, 

We will be adding the `indexes` property to the mongoose adapter in the near term. If you don't want to wait, you need to follow the migrations documentation that should help you add indexes to your models.

You'll need to create the migration, see https://payloadcms.com/docs/database/migrations.

Use Payload to access your models and call `payload.db.collections[collectionSlug].createIndexes({ //...`
https://mongoosejs.com/docs/api/model.html#Model.createIndexes()

You will commit your migration and then you will need to run the migration on your production instance or other environments that depend on the indexes being there.

I hope this helps!

**Comment by blueMountain007** [comment]
@DanRibbens  Is the indexes property supported in the new version? If it is already supported, please let me know here, thank you.

**Comment by DanRibbens** [comment]
I've converted this to a discussion as I don't think we are going to add index creation back to the config like we had done in 1.x

If you don't want to deal with db migrations you also have the option of adding custom index creating to the config in the `onInit` function.

In your project or a plugin, you can create indexes using:

```ts
// payload.config.ts
   // ..
	onInit: ({payload}) => {
		payload.db.collections[collectionSlug].index({ fields: { carMake: 1, name: 1 }, options: { unique: true }})
		payload.db.collections[collectionSlug]index({ fields: {carMake: 1, fullname: 1 }, options: { unique: true }})
	}),
```

This can also be done in plugins that add collections that need to also specify indexes.
